servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -418,19 +445,15 @@ protected Hash doOps(
     }
   }
 
-  private static CommitMeta meta(Principal principal, CommitMeta commitMeta)
-      throws NessieConflictException {
-    if (commitMeta.getCommitter() != null) {
-      throw new NessieConflictException(
-          "Cannot set the committer on the client side. It is set by the server.");
-    }
-    String committer = principal == null ? "" : principal.getName();
+  private CommitMeta meta(CommitMeta commitMeta, Map<String, String> additionalProps) {
+    String committer = Optional.ofNullable(getPrincipal()).map(p -> p.getName()).orElse("");
```

Comment:
Rather leave the ternary as it was before.

Reply:
Should Principal be supplied as a parameter? If yes, why since it is available as a class level function?

I can move the implementation to following variant if you feel that's easier to read - 
```
    Principal principal = getPrincipal();
    String committer = principal == null ? "" : principal.getName();
```